features/FEATURE_BLE/targets/TARGET_CORDIO/source/CordioPalSecurityManager.cpp

```diff
@@ -0,0 +1,601 @@
+    bool &enabled
+) {
+    return BLE_ERROR_NOT_IMPLEMENTED;
+}
+
+////////////////////////////////////////////////////////////////////////////
+// Security settings
+//
+
+ble_error_t CordioSecurityManager::set_authentication_timeout(
+    connection_handle_t, uint16_t timeout_in_10ms
+) {
+    return BLE_ERROR_NOT_IMPLEMENTED;
+}
+
+ble_error_t CordioSecurityManager::get_authentication_timeout(
+    connection_handle_t, uint16_t &timeout_in_10ms
+) {
+    return BLE_ERROR_NOT_IMPLEMENTED;
+}
+
+ble_error_t CordioSecurityManager::slave_security_request(
+    connection_handle_t connection,
+    AuthenticationMask authentication
+) {
+    DmSecSlaveReq(connection, authentication.value());
+    return BLE_ERROR_NONE;
+}
+
+////////////////////////////////////////////////////////////////////////////
+// Encryption
+//
+
+ble_error_t CordioSecurityManager::enable_encryption(
+    connection_handle_t connection,
+    const ltk_t &ltk,
+    const rand_t &rand,
+    const ediv_t &ediv
+) {
+    dmSecLtk_t sec_ltk;
+    memcpy(sec_ltk.key, ltk.data(), ltk.size());
+    memcpy(sec_ltk.rand, rand.data(), rand.size());
+    memcpy(&sec_ltk.ediv, ediv.data(), ediv.size());
+
+    DmSecEncryptReq(
+        connection,
+        /* FIXME: Security Level */ DM_SEC_LEVEL_ENC_AUTH,
+        &sec_ltk
+    );
+
+    return BLE_ERROR_NONE;
+}
+
+ble_error_t CordioSecurityManager::enable_encryption(
+    connection_handle_t connection,
+    const ltk_t &ltk
+) {
+    dmSecLtk_t sec_ltk = { 0 };
+    memcpy(sec_ltk.key, ltk.data(), ltk.size());
+
+    DmSecEncryptReq(
+        connection,
+        DM_SEC_LEVEL_ENC_LESC,
+        &sec_ltk
+    );
+
+
+    return BLE_ERROR_NONE;
+}
+
+ble_error_t CordioSecurityManager::disable_encryption(connection_handle_t connection)
+{
+    return BLE_ERROR_NOT_IMPLEMENTED;
+}
+
+ble_error_t CordioSecurityManager::get_encryption_status(
+    connection_handle_t connection, LinkSecurityStatus_t &status
+) {
+    return BLE_ERROR_NOT_IMPLEMENTED;
+}
+
+ble_error_t CordioSecurityManager::get_encryption_key_size(
+    connection_handle_t, uint8_t &bitsize
+) {
+    return BLE_ERROR_NOT_IMPLEMENTED;
+}
+
+ble_error_t CordioSecurityManager::encrypt_data(
+    const key_t &key,
+    encryption_block_t &data
+) {
+    return BLE_ERROR_NOT_IMPLEMENTED;
+}
+
+////////////////////////////////////////////////////////////////////////////
+// Privacy
+//
+
+ble_error_t CordioSecurityManager::set_private_address_timeout(
+    uint16_t timeout_in_seconds
+) {
+    DmPrivSetResolvablePrivateAddrTimeout(timeout_in_seconds);
+    return BLE_ERROR_NONE;
+}
+
+////////////////////////////////////////////////////////////////////////////
+// Keys
+//
+
+ble_error_t CordioSecurityManager::set_ltk(
+    connection_handle_t connection, const ltk_t& ltk
+) {
+    // FIXME: get access to the security level of a key
+    DmSecLtkRsp(
+        connection,
+        /* key found */ true,
+        /* sec level ??? */ DM_SEC_LEVEL_ENC_AUTH,
```

Comment:
Would it be possible to add the security level as a parameter ?

Reply:
this has been resolved by adding the information as bools

Reply:
I've updated the pal accordingly.